pkg/kube/kube_handler_test.go

```diff
@@ -1088,137 +1093,523 @@ func setClientsetData(cs fakeCombinedClientset, namespaceNames []existingNs, err
+					},
+					Header: http.Header{},
+					Host:   "example.com",
+				},
+			},
 		},
 		{
-			name:             "it includes the auth creds",
-			requestNamespace: kubeappsNamespace,
-			requestData:      `{"appRepository": {"name": "test-repo", "repoURL": "http://example.com/test-repo", "authHeader": "test-me"}}`,
-			expectedURLs:     []string{"http://example.com/test-repo/index.yaml"},
-			expectedHeaders:  http.Header{"Authorization": []string{"test-me"}},
+			name: "it returns an OCI validator for an OCI repo",
+			appRepo: &v1alpha1.AppRepository{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-repo",
+				},
+				Spec: v1alpha1.AppRepositorySpec{
+					URL:             "http://example.com/test-repo",
+					Type:            "oci",
+					OCIRepositories: []string{"apache", "jenkins"},
+				},
+			},
+			expectedValidator: HelmOCIValidator{
+				AppRepo: &v1alpha1.AppRepository{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-repo",
+					},
+					Spec: v1alpha1.AppRepositorySpec{
+						URL:             "http://example.com/test-repo",
+						Type:            "oci",
+						OCIRepositories: []string{"apache", "jenkins"},
+					},
+				},
+			},
 		},
 		{
-			name:             "validation fails if docker registry secrets included for a global repo",
-			requestNamespace: kubeappsNamespace,
-			requestData:      `{"appRepository": {"name": "test-repo", "repoURL": "http://example.com/test-repo", "registrySecrets": ["secret-1"]}}`,
-			expectedError:    ErrGlobalRepositoryWithSecrets,
+			name: "it returns an error for an OCI repo if no repositories are given",
+			appRepo: &v1alpha1.AppRepository{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-repo",
+				},
+				Spec: v1alpha1.AppRepositorySpec{
+					URL:             "http://example.com/test-repo",
+					Type:            "oci",
+					OCIRepositories: []string{},
+				},
+			},
+			expectedError: ErrEmptyOCIRegistry,
 		},
+	}
+
+	cmpOpts := []cmp.Option{cmpopts.IgnoreUnexported(http.Request{})}
+	cmpOpts = append(cmpOpts, cmpopts.IgnoreFields(http.Request{}, "Proto", "ProtoMajor", "ProtoMinor"))
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			httpValidator, err := getValidator(tc.appRepo)
+			if got, want := err, tc.expectedError; got != want {
+				t.Fatalf("got: %+v, want: %+v", err, tc.expectedError)
+			} else if tc.expectedError != nil {
+				return
+			}
+
+			if got, want := httpValidator, tc.expectedValidator; !cmp.Equal(got, want, cmpOpts...) {
+				t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got, cmpOpts...))
+			}
+		})
+	}
+}
+
+func TestNonOCIValidate(t *testing.T) {
+	validRequest, err := http.NewRequest("GET", "http://example.com/index.yaml", strings.NewReader(""))
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	testCases := []struct {
+		name             string
+		httpValidator    HelmNonOCIValidator
+		fakeHttpError    error
+		fakeRepoResponse *http.Response
+		expectedResponse *ValidationResponse
```

Comment:
This one for testing the non-OCI validation. It's just unit-testing the `Validate` method only: for a given upstream response, what's the validation response to the client.